packages/smart-wallet/src/walletFactory.js

```diff
@@ -27,10 +27,14 @@ export const meta = {
     { storageNode: M.eref(M.remotable('StorageNode')) },
     { walletBridgeManager: M.eref(M.remotable('walletBridgeManager')) },
   ),
-  upgradability: 'canUpgrade',
+  // UNTIL this compat stuff lands???@@@
+  // upgradability: 'canUpgrade',
 };
 harden(meta);
 
+// advertise privateArgsShape to old Zoe too
+export const privateArgsShape = meta.privateArgsShape;
```

Comment:
@turadg I'm getting a [ci failure](https://github.com/Agoric/agoric-sdk/actions/runs/5873123926/job/15925980527?pr=8147#step:14:30) in a test that `privateArgsShape` is enforced.

Help me understand how to get this contract compatible with mainnet and with master?

Reply:
That's saying,
```
not ok 7 - startWalletFactory › customTermsShape %ava-dur=2333ms
…
Error#1: privateArgs: {} - Must have missing properties ["storageNode"]
```
The cause is that the test is expecting an error on customTerms but you only re-exported the `privateArgsShape`.

The simplest thing is to just stick to the old way. I've pushed a commit for that to run CI on it. I've also pushed a commit to make the use of `prepare` not require all the types to change to it.